src/test/java/pl/coderslab/imageManager/services/ImageServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package pl.coderslab.imageManager.services;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.commons.logging.Logger;
+import org.junit.platform.commons.logging.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import pl.coderslab.imageManager.model.ImageDescription;
+import pl.coderslab.imageManager.repositories.ImageRepository;
+
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReader;
+import javax.imageio.metadata.IIOMetadata;
+import javax.imageio.stream.ImageInputStream;
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+
+class ImageServiceTest {
```

Comment:
Ogólnie cała ta klasa będzie do przepisania, spójrz na komentarze niżej i już Ci mówię jak napisać dobre testy do tego:

1. Powinnaś stworzyć klasę która będzie symulowała działanie bazy danych, przykładowo: 
![image](https://user-images.githubusercontent.com/18016435/111885722-d92d7980-89c9-11eb-9183-d711c548d758.png)

Obecne użycie mocków jest złe. 

2. Ta klasa powinna działać tak, że pod spodem ma tę hashMapę, która jest symulacją bazy danych. Chodzi o to żeby faktycznie te obiekty w testach gdzieś zapisywać i w tym przypadku to będzie właśnie ta hashMapa.

3. W klasie pojawi Ci się duuużo metod, należy nadpisać tylko te które są wykorzystywane w testach. Np. jeśli używasz save to powinnaś w tej klasie nadpisać metodę save. Jak to zrobić, jak np. nadpisać save? Wtedy taka metoda powinna po pierwsze wygenerować nowy klucz dla obiektu imageDescription (jak zapisujesz do bazy to Ci baza generuje klucz, tu nie ma prawdziwej bazy tylko jest hashMapa, a zatem sama musisz to zrobić i przydizelić np. następną wartość id której jeszcze nie ma w hashmapie). No i potem po prostu robisz .add, do hashmapy i tyle - masz już zapis do bazy. Z czytaniem jest prościej bo po prostu się robi find czy tam get na tej hashmapie i masz obiekt. 

4. I musisz tę testową implementację repozytorium przekazywać do serwisów itd. tak żeby faktycznie zapisywało do tej hashmapy. 

No i tyle, czyli najogólniej rzecz ujmując chodzi o to żeby zasymulować działanie bazy przy pomocy hashmapy która jest pod spodem i żeby te obiekty faktycznie gdzieś były zapisywane, bo w przypadku mocków nic się z nimi nie dzieje :)